frontend/src/lib/lemon-ui/LemonFileInput/LemonFileInput.tsx

```diff
@@ -61,9 +61,11 @@ export const LemonFileInput = ({
 
         const eventFiles = e.target.files
         const filesArr = Array.prototype.slice.call(eventFiles)
-        const localFiles = multiple ? [...files, ...filesArr] : [filesArr[0]]
-        setFiles(localFiles)
-        onChange?.(localFiles)
+        const localFiles = multiple ? [...files, ...filesArr] : filesArr.slice(0, 1)
+        if (filesArr.length > 0) {
+            setFiles(localFiles)
+            onChange?.(localFiles)
+        }
```

Comment:
Nit: `localFiles` isn't used outside of the if statement so it should be in it. 
```suggestion
        if (filesArr.length > 0) {
            const localFiles = multiple ? [...files, ...filesArr] : filesArr.slice(0, 1)
            setFiles(localFiles)
            onChange?.(localFiles)
        }
```